src/Microsoft.AspNet.OData.Shared/Formatter/ODataOutputFormatterHelper.cs

```diff
@@ -349,5 +348,62 @@ private static bool IsOperationPath(ODataPath path)
 
             return false;
         }
+
+        private static Microsoft.OData.UriParser.ODataPath ConvertPath(ODataPath path)
+        {
+            if (path == null)
+            {
+                return null;
+            }
+
+            if (IsOperationPath(path))
```

Comment:
What is the value in doing this check? It's going to walk the path looking for an operationSegment or operationImportSegment, and if it returns true, we're going to go to the last segment and check to see if it's an operationSegment or operationImportSegment.  Can we remove the IsOperationPath(path) check and just rely on the existing checks on line 363 and 369?  The only difference I can see in doing the check first is that the current logic will return a null if there is an operation (i.e., a composible function) in the path but not as the last segment -- is that desired behavior?

Reply:
I'd like to keep the behavior unchanged. if we consider the compossible function, the logic here is not enough to cover it.  

---
In reply to: [401841900](https://github.com/OData/WebApi/pull/2060#discussion_r401841900) [](ancestors = 401841900)